ibverbs/src/lib.rs

```diff
@@ -1488,26 +1530,16 @@ impl<'res> QueuePair<'res> {
     ///
     /// [1]: http://www.rdmamojo.com/2013/01/26/ibv_post_send/
     #[inline]
-    pub unsafe fn post_send<T, R>(
+    pub unsafe fn post_send<'a>(
         &mut self,
-        mr: &mut MemoryRegion<T>,
-        range: R,
+        local: &[LocalMemorySlice<'a>],
         wr_id: u64,
-    ) -> io::Result<()>
-    where
-        R: sliceindex::SliceIndex<[T], Output = [T]>,
-    {
-        let range = range.index(mr);
-        let mut sge = ffi::ibv_sge {
-            addr: range.as_ptr() as u64,
-            length: mem::size_of_val(range) as u32,
-            lkey: (*mr.mr).lkey,
-        };
+    ) -> io::Result<()> {
         let mut wr = ffi::ibv_send_wr {
             wr_id,
             next: ptr::null::<ffi::ibv_send_wr>() as *mut _,
-            sg_list: &mut sge as *mut _,
-            num_sge: 1,
+            sg_list: local.as_ptr() as *mut ffi::ibv_sge,
```

Comment:
I don't think this works at the moment (or if so, it's accidentaly), because it requires that `LocalMemorySlice` has exactly the same layout as `ibv_sge`. To fix this you'd have to add `#[repr(transparent)]` to `LocalMemorySlice`.